mesonbuild/interpreter/interpreter.py

```diff
@@ -1706,12 +1709,13 @@ def func_vcs_tag(self, node: mparser.BaseNode, args: T.List['TYPE_var'], kwargs:
                 source_dir,
                 replace_string,
                 regex_selector] + vcs_cmd,
-            'input': kwargs['input'],
-            'output': kwargs['output'],
-            'build_by_default': True,
-            'build_always_stale':True,
-        }
-        return self._func_custom_target_impl(node, [kwargs['output']], cmd_kwargs)
+            self.source_strings_to_files(kwargs['input'], strict=False),
```

Comment:
This `strict=False` hack is unique to `custom_target()` because we document passing a typelib file to custom_target as a manually crafted string, see commit 1e5d7f212246e967f3473a55f632f6f5eda87b65 for details.

This seems like an excellent opportunity to reduce the surface area of that hack, so that functions such as vcs_tag which happen to create a `build.CustomTarget` no longer get this silliness as a side effect of using `_func_custom_target_impl`.